### PR TITLE
Drupal data loader fixes

### PR DESCRIPTION
**Pull request by bderusha**
[ECOM-5510](https://openedx.atlassian.net/browse/ECOM-5510)

@edx/ecommerce please review

### CODE REVIEW

File: course_discovery/apps/course_metadata/data_loaders/marketing_site.py [hunk 1]
```diff
@@ -369,15 +369,21 @@ def process_node(self, data):
...
             'video': self.get_video(data),
             'short_description': self.clean_html(data['field_course_sub_title_short']),
             'level_type': self.get_level_type(data['field_course_level']),
             'card_image_url': self._get_nested_url(data.get('field_course_image_promoted')),
         }
-        course, __ = Course.objects.update_or_create(key__iexact=key, partner=self.partner, defaults=defaults)
+        course, created = Course.objects.get_or_create(key__iexact=key, partner=self.partner, defaults=defaults)
+
+        published = self.get_course_run_status(data) == CourseRun.Status.Published
+        if not created and published:
```
Comment (clintonb) [inline course_discovery/apps/course_metadata/data_loaders/marketing_site.py:386]: What's this?

Comment (bderusha) [reply course_discovery/apps/course_metadata/data_loaders/marketing_site.py:386]: If the course already exists (aka `not created`) update the fields only if the `course_run` we got from drupal is published.

One issue we had was that people were putting temp data into required drupal fields (e.g. `.`) for unpublished courses.  We were overwriting the course info with this data regardless of the `course_run` published state.

We could omit the `created` check but if it was created we'd be resaving the values used on create. Seemed like a waste

Comment (clintonb) [reply course_discovery/apps/course_metadata/data_loaders/marketing_site.py:386]: People are weird.

**Comment by clintonb** [comment]
👍

File: course_discovery/apps/course_metadata/data_loaders/marketing_site.py (hunk 1 above)
Comment (schenedx) [reply course_discovery/apps/course_metadata/data_loaders/marketing_site.py:386]: #nitpick @bderusha I also benefited from the explanation on this condition. Please write a comment in code on why we need this. It really helps to maintain the code.

**Comment by schenedx** [comment]
Just a request to write comment, rest LGTM 👍

File: course_discovery/apps/course_metadata/data_loaders/marketing_site.py [hunk 2]
```diff
@@ -369,15 +369,24 @@ def process_node(self, data):
...
             'card_image_url': self._get_nested_url(data.get('field_course_image_promoted')),
         }
-        course, __ = Course.objects.update_or_create(key__iexact=key, partner=self.partner, defaults=defaults)
+        course, created = Course.objects.get_or_create(key__iexact=key, partner=self.partner, defaults=defaults)
+
+        # If the course already exists update the fields only if the course_run we got from drupal is published.
+        # People often put temp data into required drupal fields for unpublished courses. We don't want to  overwrite
+        # the course info with this data, so we only update course info from published sources.
+        published = self.get_course_run_status(data) == CourseRun.Status.Published
+        if not created and published:
```
Comment (rlucioni) [inline course_discovery/apps/course_metadata/data_loaders/marketing_site.py:386]: @bderusha really minor, but this reads like an English sentence if reversed: `if published and not created:`.

**Comment by rlucioni** [comment]
👍

**Pull request merged**